src/checker/http_checker.rs

```diff
@@ -489,6 +526,112 @@ mod tests {
         let trace_header_sampling = make_trace_header(&config_with_sampling, &trace_id, span_id);
         assert_eq!(trace_header_sampling, format!("{}-{}", trace_id, span_id));
     }
-    // TODO: Figure out how to simulate a DNS failure
-    // assert_eq!(check_domain(&client, "https://hjkhjkljkh.io/".to_string()).await, CheckResult::FAILURE(FailureReason::DnsError("failed to lookup address information: nodename nor servname provided, or not known".to_string())));
+
+    #[tokio::test]
+    #[cfg(target_os = "linux")]
+    async fn test_ssl_errors_linux() {
+        // Helper function to create various bad certificates
+        fn create_bad_cert(cert_type: &str) -> (Vec<u8>, Vec<u8>) {
```

Comment:
maybe make a little enum because why not

Reply:
in the tests? is that really necessary?

Reply:
I mean I think test code is just as important as production code. There's no overhead to adding an enum or anything like that. I just tend to really dislike magic strings.

```rs
enum TestCertType {
    Expired,
    WrongHost,
	SelfSigned,
}
```

anyway more of a nit, not too important

Reply:
i've updated the tests to use an Enum.